asf_tools/composite.py

```diff
@@ -113,31 +99,24 @@ def reproject_to_median_utm(files, pol, resolution=None):
         pix_size = get_max_pixel_size(files)
         logging.info(f"Using maximum pixel size {pix_size}")
 
-    # Get the median UTM zone and hemisphere
-    home_zone = np.median(parse_zones(files))
-    logging.info(f"Home zone is {home_zone}")
-    hemi = get_hemisphere(files[0])
-    logging.info(f"Hemisphere is {hemi}")
+    target_epsg_code = get_target_epsg_code(files)
+    logging.info(f"Target EPSG code is {target_epsg_code}")
 
     # Reproject files as needed
     logging.info("Checking projections")
     new_files = []
     for fi in files:
         fi_name = os.path.split(fi)[1]
-        my_zone = get_zone_from_proj(fi)
+        my_epsg_code = get_epsg_code(fi)
         name = fi_name.replace(".tif", "_reproj.tif")
         afi = fi.replace(f"_{pol}.tif", "_area.tif")
         aname = fi_name.replace(f"_{pol}.tif", "_area_reproj.tif")
         if not os.path.isfile(name):
             x, y, trans, proj = saa.read_gdal_file_geo(saa.open_gdal_file(fi))
-            if my_zone != home_zone:
+            if my_epsg_code != target_epsg_code:
```

Comment:
I suspect this fixes a bug, previously if your input files were say `EPSG:32606` and `EPSG:32706` (same zone different hemisphere) they may not have been projected to a common format.